dts/bindings/timer/andestech,atcpit100.yaml

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2022, Andes Technology Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description:
+    This is a representation of the Andes Technology atcpit100 PIT nodes
+
+compatible: "andestech,atcpit100"
+
+include: base.yaml
+
+properties:
+    reg:
+      required: true
+
+    interrupts:
+      required: true
+
+    clock-frequency:
+      type: int
+      description: channel clock source
+      required: true
+
+    prescaler:
+      type: int
+      default: 1
+      description: Prescaler value determines frequency (clock-frequency/prescaler)
```

Comment:
If you are going to use `default:`, you must justify the choice of why you picked the default value of 1 in the description.

- do not justify the choice in a github comment; it must be in the description
- do not just restate the default value; this is already in the binding. Instead, justify **why** the value is the way it is

Additional rules apply; see https://docs.zephyrproject.org/latest/build/dts/bindings.html#rules-for-default-values

Reply:
@mbolivar-nordic out of curiosity is there any preference between (1) having a `default` in the `yaml` file vs (2) using `DT_PROP_OR` in the code?

Reply:
@carlocaione we don't have a hard policy or preference.

My advice would be to prefer `default` in cases where it makes sense, and especially when the default value is not 'zero' (for whatever value of 'zero' makes sense depending on the property type, e.g. an empty array could be 'zero'). In that case, the driver's behavior will be documented more clearly in the binding. So for example if you have a nonzero default `frame-delay` for some I/O driver, that would make a good `default`.

But if a missing optional property has a completely obvious behavior when it's not specified, `DT_FOO_OR` makes sense. So like a missing `foo-gpios` property is usually handled with `GPIO_DT_SPEC_INST_GET_OR(..., {0})`, because that's kind of obviously what the driver 'ought' to do.

Reply:
I know I prefer to push people to using `DT_PROP_OR`.  I wonder if we should look to phase out `default`.  I can't remember if we had `DT_PROP_OR` at the time, but it feels like `default` is more difficult for people to use, and I'm not sure of the benefit.

Reply:
> I'm not sure of the benefit.

Documentation in the bindings index is a big win IMO

Reply:
Agreed.  I started a thread on discord #devicetree to see if we can make some progress on defaults.  I glanced at a handful of cases in tree and all of the ones I looked at didn't have ANY explanation as to why the default :(.

Reply:
> all of the ones I looked at didn't have ANY explanation as to why the default :(.

Ugh. That's a bug, we do have documented rules about this and I always -1 when I see it

Reply:
@mbolivar-nordic  Thank you for your comments, I updated the description in this property.